src/wovnio/wovnphp/Store.php

```diff
@@ -125,7 +126,11 @@ public function updateSettings($vals)
 
         // getting the url pattern
         if (isset($vals['url_pattern_name']) && $vals['url_pattern_name'] === 'query') {
-            $vals['url_pattern_reg'] = '((\?.*&)|\?)wovn=(?P<lang>[^&]+)(&|$)';
+            if (isset($vals['lang_param_name'])) {
```

Comment:
internally, can we make it so lang_param_name is always set to `wovn` by default? That just gets overridden if the user sets it? So we could remove this else branch.

Reply:
Ugh, I agree this branch is ugly. The predicament comes from how the default values are set somewhere else and not accessed in this function, but we kind of need it for creating `url_pattern_reg`. The default value for the setting always set to `wovn`, but it is only merged after this function has returned.

I will not be doing any settings management refactoring here, but I can try to make branch a bit cleaner.

Reply:
I effectively just moved the branch, and I think it got better. It still is ugly, but I want to work within the frame in which all the other settings are managed.